test/Microsoft.AspNetCore.HostFiltering.Tests/HostFilteringMiddlewareTests.cs

```diff
@@ -0,0 +1,310 @@
+        public async Task AllowsMissingHost(bool allowed, int status)
+        {
+            var builder = new WebHostBuilder()
+                .ConfigureServices(services =>
+                {
+                    services.AddHostFiltering(options =>
+                    {
+                        options.AllowEmptyHosts = allowed;
+                        options.AllowedHosts.Add("Localhost");
+                    });
+                })
+                .Configure(app =>
+                {
+                    app.Use((ctx, next) =>
+                    {
+                        ctx.Request.Headers.Remove(HeaderNames.Host);
+                        return next();
+                    });
+                    app.UseHostFiltering();
+                    app.Run(c =>
+                    {
+                        Assert.False(c.Request.Headers.TryGetValue(HeaderNames.Host, out var host));
+                        return Task.CompletedTask;
+                    });
+                });
+            var server = new TestServer(builder);
+            var response = await server.CreateClient().GetAsync("/");
+            Assert.Equal(status, (int)response.StatusCode);
+        }
+
+        [Theory]
+        [InlineData(true, 200)]
+        [InlineData(false, 400)]
+        public async Task AllowsEmptyHost(bool allowed, int status)
+        {
+            var builder = new WebHostBuilder()
+                .ConfigureServices(services =>
+                {
+                    services.AddHostFiltering(options =>
+                    {
+                        options.AllowEmptyHosts = allowed;
+                        options.AllowedHosts.Add("Localhost");
+                    });
+                })
+                .Configure(app =>
+                {
+                    app.Use((ctx, next) =>
+                    {
+                        ctx.Request.Headers[HeaderNames.Host] = " ";
+                        return next();
+                    });
+                    app.UseHostFiltering();
+                    app.Run(c =>
+                    {
+                        Assert.True(c.Request.Headers.TryGetValue(HeaderNames.Host, out var host));
+                        Assert.True(StringValues.Equals(" ", host));
+                        return Task.CompletedTask;
+                    });
+                    app.Run(c => Task.CompletedTask);
+                });
+            var server = new TestServer(builder);
+            var response = await server.CreateClient().GetAsync("/");
+            Assert.Equal(status, (int)response.StatusCode);
+        }
+
+        [Theory]
+        [InlineData("localHost", "localhost")]
+        [InlineData("localHost", "*")] // Any - Used by HttpSys
+        [InlineData("localHost", "[::]")] // IPv6 Any - This is what Kestrel reports when binding to *
+        [InlineData("localHost", "0.0.0.0")] // IPv4 Any
+        [InlineData("localhost:9090", "example.com;localHost")]
```

Comment:
Where is the example.com coming from here?

Reply:
example.com is a separate allowed host. I wanted to make sure it would test the whole list.